kinder/ci/tools/update-workflows/templates/workflows/rootless-tasks.yaml

```diff
@@ -0,0 +1,277 @@
+          echo "kube-scheduler is running as user kubeadm-ks"
+      else
+          echo "ERROR: kube-scheduler is not running as user kubeadm-ks"
+          res=1
+      fi
+
+      if pgrep etcd | xargs ps o user:16 --no-headers -p | grep -q kubeadm-etcd ; then
+          echo "etcd is running as user kubeadm-etcd"
+      else
+          echo "ERROR: etcd is not running as user kubeadm-etcd"
+          res=1
+      fi
+
+      if pgrep etcd | xargs ps o group:16 --no-headers -p | grep -q kubeadm-etcd ; then
+          echo "etcd is running as user kubeadm-etcd"
+      else
+          echo "ERROR: etcd is not running as user kubeadm-etcd"
+          res=1
+      fi
+
+      if [[ "\${res}" = 0 ]]; then
+        echo "All verify checks passed, congrats!"
+        echo ""
+      else
+        echo "One or more verify checks failed! See output above..."
+        echo ""
+        exit 1
+      fi
+      EOF
+
+      chmod +x /tmp/verify-rootless.sh
+- name: copy verify-rootless.sh on controlplane nodes
+  cmd: kinder
+  args:
+    - cp
+    - --name={{ .vars.clusterName }}
+    - /tmp/verify-rootless.sh
+    - "@cp*:/kinder/verify-rootless.sh"
+    - --loglevel=debug
+- name: init
+  description: |
+    Initializes the Kubernetes cluster with version "initVersion"
+    by starting the boostrap control-plane nodes
+  cmd: kinder
+  args:
+    - do
+    - kubeadm-init
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+    - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+    - --kubeadm-feature-gate="RootlessControlPlane=true"
+  timeout: 5m
+- name: join
+  description: |
+    Join the other nodes to the Kubernetes cluster
+  cmd: kinder
+  args:
+    - do
+    - kubeadm-join
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+    - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+  timeout: 10m
+- name: run verify-rootless.sh on controlplane nodes before upgrades
+  cmd: kinder
+  args:
+    - exec
+    - --name={{ .vars.clusterName }}
+    - "@cp*"
+    - /kinder/verify-rootless.sh
+    - --loglevel=debug
+- name: e2e-kubeadm
+  description: |
+    Runs kubeadm e2e tests
+  cmd: kinder
+  args:
+    - test
+    - e2e-kubeadm
+    - --test-flags=--report-dir={{ .env.ARTIFACTS }} --report-prefix=e2e-kubeadm
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+  timeout: 10m
+- name: e2e
+  description: |
+    Runs Kubernetes e2e test (conformance)
+  cmd: kinder
+  args:
+    - test
+    - e2e
+    - --test-flags=--report-dir={{ .env.ARTIFACTS }} --report-prefix=e2e
+    - --parallel
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+  timeout: 35m
+- name: upgrade
+  description: |
+    upgrades the cluster to Kubernetes "upgradeVersion"
+  cmd: kinder
+  args:
+    - do
+    - kubeadm-upgrade
+    - --upgrade-version={{ .vars.kubernetesVersion }}
+    - --name={{ .vars.clusterName }}
+    - --loglevel=debug
+    - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
```

Comment:
did you check if this works correctly, locally?
i had some concerns around artifact preparation and same-to-same version upgrades.

Reply:
When I run it locally it does not use the local version of kinder and my changes aren't showing up. Do you know why that is? It builds kinder locally but it complains with --kubeadm-feature-gate is not a valid flag.

Reply:
hm, it should work because the script is building from your local clone.

https://github.com/kubernetes/kubeadm/blob/master/kinder/ci/kinder-run.sh#L62

you could add `whereis kinder` there and see where it comes from.
perhaps you have multiple binaries on disk, somehow.

Reply:
i must admit, i usually do:

```
cd kinder
go build
kinder test workflow path-to.yaml .... flags
```

Reply:
Ah that was it! It was using kinder from another location. Thank you!